src/Microsoft.Health.Dicom.Core/Features/Store/LoggingIndexDataStore.cs

```diff
@@ -261,5 +273,45 @@ public async Task<int> IncrementDeletedInstanceRetryAsync(VersionedInstanceIdent
                 throw;
             }
         }
+
+        public async Task<int> RetrieveNumDeletedExceedRetryCountAsync(int maxRetryCount, CancellationToken cancellationToken)
+        {
+            LogRetrieveNumDeletedExceedRetryCountAsyncDelegate(_logger, maxRetryCount, null);
+
+            try
+            {
+                int returnValue = await _indexDataStore.RetrieveNumDeletedExceedRetryCountAsync(maxRetryCount, cancellationToken);
+
+                LogOperationSucceededDelegate(_logger, null);
+
+                return returnValue;
+            }
+            catch (Exception ex)
+            {
+                LogOperationFailedDelegate(_logger, ex);
+
+                throw;
+            }
+        }
+
+        public async Task<DateTimeOffset> GetOldestDeletedAsync(CancellationToken cancellationToken)
```

Comment:
Should we return null when there is no deleted instance?

Reply:
Keeping for metrics as discussed offline